From 16809870bc952109995c43c034d8028a7fba49a7 Mon Sep 17 00:00:00 2001
From: brandnkdavis <brandnkdavis@gmail.com>
Date: Fri, 15 May 2026 13:57:30 +0000
Subject: [PATCH 10/22] sepolicy: qti: Address some subsystem_daemon denials

Change-Id: I33b80fbf88fcc9ceeb24979aa35b5bfa1e1d2ec2
(cherry picked from commit ac5f704208dd19f4a33ccb000c74c37ccedc663b)

Ledger: hardware_oplus.md Accepted Group A row 4 (ac5f7042)
Change-Id: I1b39272dbb47a6653cf1c9e303dc42b704d703f3
(cherry picked from commit ec475bdf9cc394597babc291b4027e921459e5fe)
(cherry picked from commit e18b69237ccc02c99d02cf8a0d83dfcb040714d9)
---
 sepolicy/qti/vendor/subsystem_daemon.te | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/sepolicy/qti/vendor/subsystem_daemon.te b/sepolicy/qti/vendor/subsystem_daemon.te
index a6d01541..8177378e 100644
--- a/sepolicy/qti/vendor/subsystem_daemon.te
+++ b/sepolicy/qti/vendor/subsystem_daemon.te
@@ -11,13 +11,20 @@ binder_call(vendor_hal_subsys_server, vendor_hal_subsys_client)
 
 hal_attribute_service(vendor_hal_subsys, hal_subsys_service)
 
-allow subsystem_daemon block_device:dir r_dir_perms;
 allow subsystem_daemon self:qipcrtr_socket create_socket_perms_no_ioctl;
+
+allow subsystem_daemon block_device:dir r_dir_perms;
+allow subsystem_daemon vendor_modem_efs_partition_device:blk_file r_file_perms;
 allow subsystem_daemon vendor_reserve_partition:blk_file r_file_perms;
+
+allow subsystem_daemon mnt_vendor_file:dir search;
+allow subsystem_daemon oplus_reserve_file_type:dir search;
+
 allow subsystem_daemon vendor_sysfs_soc_sensitive:file r_file_perms;
 
 set_prop(subsystem_daemon, vendor_nw_exported_system_prop)
 set_prop(subsystem_daemon, vendor_radio_prop)
 
 r_dir_file(subsystem_daemon, vendor_proc_engineer)
+rw_dir_file(subsystem_daemon, oplus_reserve_radio_file_type)
 rw_dir_file(subsystem_daemon, vendor_olog_file)
